core/src/apps/cardano/layout.py

```diff
@@ -130,6 +145,48 @@ async def confirm_transaction(ctx, amount: int, fee: int, protocol_magic: int):
     await require_hold_to_confirm(ctx, Paginated([t1, t2]))
 
 
+async def confirm_certificate(
+    ctx: wire.Context, certificate: CardanoTxCertificateType
+) -> bool:
+    pages = []
+
+    t1 = Text("Confirm transaction", ui.ICON_SEND, ui.GREEN)
+    t1.normal("Confirm:")
+    t1.bold(CERTIFICATE_TYPE_NAMES[certificate.type])
+    t1.normal("for account:")
+    t1.bold(address_n_to_str(to_account_path(certificate.path)))
+    pages.append(t1)
+
+    if certificate.type == CardanoCertificateType.STAKE_DELEGATION:
+        t2 = Text("Confirm transaction", ui.ICON_SEND, ui.GREEN)
+        t2.normal("to pool:")
+        t2.bold(hexlify(certificate.pool).decode())
+        pages.append(t2)
+
+    await require_confirm(ctx, Paginated(pages))
+
+
+async def confirm_withdrawal(
+    ctx: wire.Context, withdrawal: CardanoTxWithdrawalType
+) -> bool:
+    t1 = Text("Confirm transaction", ui.ICON_SEND, ui.GREEN)
+    t1.normal("Confirm withdrawal")
+    t1.normal("for account:")
+    t1.bold(address_n_to_str(to_account_path(withdrawal.path)))
+    t1.normal("Amount:")
+    t1.bold(format_coin_amount(withdrawal.amount))
+
+    await require_confirm(ctx, t1)
+
+
+async def confirm_metadata_hash(ctx: wire.Context, metadata_hash: bytes) -> bool:
+    t1 = Text("Confirm transaction", ui.ICON_SEND, ui.GREEN)
+    t1.normal("Confirm metadata hash:")
+    t1.bold(hexlify(metadata_hash).decode())
+
+    await require_confirm(ctx, t1)
```

Comment:
It doesn't make sense to show information which does not make any sense to user. Let's drop this confirm dialog for now.

Reply:
Also done [here](https://github.com/trezor/trezor-firmware/pull/1145/commits/1bcf5a81ebf3b0df03b25766cd7f94749233fcbf).

Reply:
This is now dead code - can we remove this function?

Reply:
Of course, sorry, I thought I had done that. Removed [here](https://github.com/trezor/trezor-firmware/pull/1145/commits/b4ee6dd45a9a15545a4bb58a9950ee3f50f76159).